runtime/near-vm-runner/src/cache.rs

```diff
@@ -292,3 +292,61 @@ pub mod wasmer1_cache {
         }
     }
 }
+
+pub(crate) fn precompile_contract_impl(
+    vm_kind: VMKind,
+    wasm_code: &ContractCode,
+    config: &VMConfig,
+    cache: Option<&dyn CompiledContractCache>,
+) -> Result<bool, ContractPrecompilatonError> {
+    let cache = match cache {
+        None => return Ok(false),
+        Some(it) => it,
+    };
+    let key = get_key(wasm_code, vm_kind, config);
+    // Check if we already cached with such a key.
+    match cache.get(&(key.0).0) {
+        // If so - do not override.
+        // TODO: is it correct?
+        Ok(Some(_)) => return Ok(false),
```

Comment:
`false` meaning is not obvious, maybe document it: Ok(true) if compile happens, Ok(false) if compile not happen. This also show `cache: Option<&dyn CompiledContractCache>` is better without `Option`, otherwise Ok(false) mean both cache hit (no compilation) and no cache (no compilation), but these two are different regarding subsequent cost

Reply:
I think this TODO is correct given key is `get_key(wasm_code, vm_kind, config)`

Reply:
Actually it's somewhat logically matching to boolean in http://www.cplusplus.com/reference/map/map/insert/, i.e. if modification of cache has happened.

Reply:
Removed TODO.